Add confirm form element which allows to confirm form fields via double-input

dontub opened the pull request:
This change requires an update of the translations. How is the pot generated?

systopia-reference: 19785

Commit:
Add confirm form element which allows to confirm form fields via double-input

bjendres commented:
> This change requires an update of the translations. How is the pot generated?

I usually run the following within the extension folder:
```
~/civi_l10n_tools/bin/create-pot-files-extensions.sh ${PWD##*/} ./ l10n
```

jensschuppe commented:
Since this is a Drupal module, using the [*Translation Template Extractor*](https://www.drupal.org/project/potx) module is the easiest way, either with *Drush* or via the user interface.

jensschuppe commented on modules/civiremote_event/src/Element/Confirm.php:
```diff
@@ -0,0 +1,118 @@
+   * {@inheritdoc}
+   */
+  public static function valueCallback(&$element, $input, FormStateInterface $form_state): array {
+    if ($input === FALSE) {
+      return $element['#default_value'] = [
+        'value1' => $element['#field']['#default_value'] ?? NULL,
+        'value2' => $element['#field']['#default_value'] ?? NULL,
+      ];
+    }
+
+    /** @phpstan-var array<string, mixed> $input */
+    return [
+      'value1' => $input['value1'] ?? $element['#field']['#default_value'] ?? NULL,
+      'value2' => $input['value2'] ?? $element['#field']['#default_value'] ?? NULL,
+    ];
```
I think it would make sense to call the `valueCallback` of the element type's render element, as there might be transformations necessary for specific field types.

dontub replied:
The value callbacks of the wrapped elements are now called. Do you agree with the implementation?

jensschuppe replied:
Maybe we should run `getValueCallback()` in the process callback of the element, so that there's validation for whether the element given really is a form element. So let's add the value callback as a `#` property in `processConfirm()` and just call it from there - what do you think?

PS: Since `valueCallback()` is being called during building the form, this comment seems obsolete.

jensschuppe commented on modules/civiremote_event/src/Element/Confirm.php:
```diff
@@ -0,0 +1,118 @@
+ * values match.
+ *
+ * Properties:
+ * - #field: The render array of the form field to confirm.
```
This should be restricted to allow form elements only, as otherwise this could take any render array, which makes validating difficult.

dontub replied:
I've renamed `#field` to `#element` so it should be more obvious that it is meant to be used for form elements.

jensschuppe replied:
See https://github.com/systopia/civiremote/pull/2#discussion_r1200272414

Commit:
Confirm: Call value callbacks of wrapped elements

elements can be confirmed.

Commit:
Confirm: Call value callbacks of wrapped elements

Rename attribute field to element to make it more clear that only form
elements can be confirmed.

Commit:
Add translation for confirm

dontub commented:
I've updated the translation to contain the confirm related strings. There were different other strings that haven't been extracted, yet. They are now part of de.po, but without translation. The way I've done the extraction made the paths to be relative to the module directory which makes the diff a little bit larger...
